src/streamlink/plugins/ard_live.py

```diff
@@ -1,56 +1,68 @@
     )
-    _livestream_schema = validate.Schema(
-        validate.xml_findall(".//assets"),
-        validate.filter(lambda x: x.attrib.get("type") != "subtitles"),
-        validate.get(0),
-        validate.xml_findall(".//asset"),
-        [validate.union({
-            "url": validate.xml_findtext("./fileName"),
-            "bitrate": validate.xml_findtext("./bitrateVideo")
-        })])
+    _mediainfo_schema = validate.Schema({
+        "mc": {
+            "_title": validate.text,
+            "_isLive": bool,
+            "_geoblocked": bool,
+            "_mediaArray": [{
+                "_mediaStreamArray": [{
+                    "_quality": validate.any(validate.text, int),
+                    "_stream": validate.any(validate.text, [validate.text]),
+                }]
+            }],
+        }
+    })
 
     @classmethod
     def can_handle_url(cls, url):
         return cls._url_re.match(url) is not None
 
     def _get_streams(self):
-        data_url = self.session.http.get(self.url, schema=self._player_url_schema)
-        if data_url:
-            res = self.session.http.get(urljoin(self.url, data_url))
-            stream_info = self.session.http.xml(res, schema=self._livestream_schema)
-
-            for stream in stream_info:
-                url = stream["url"]
-                try:
-                    if ".m3u8" in url:
-                        for s in HLSStream.parse_variant_playlist(self.session, url, name_key="bitrate").items():
-                            yield s
-                    elif ".f4m" in url:
-                        for s in HDSStream.parse_manifest(self.session, url, pvswf=self.swf_url, is_akamai=True).items():
-                            yield s
-                    elif ".mp4" in url:
-                        yield "{0}k".format(stream["bitrate"]), HTTPStream(self.session, url)
-                except IOError as err:
-                    self.logger.warning("Error parsing stream: {0}", err)
+        res = self.session.http.get(self.url)
+        data_url = self._player_url_schema.validate(res.text)
+        data_url = urljoin(res.url, data_url)
+        log.debug("Player URL: '{0}'", data_url)
+        res = self.session.http.get(data_url)
+        mediainfo = self._mediainfo_schema.validate(json.loads(res.content))
+        log.debug("Mediainfo: {0}", json.dumps(mediainfo))
+
+        for media in mediainfo["mc"]["_mediaArray"]:
+            for stream in media["_mediaStreamArray"]:
+                stream_ = stream["_stream"]
+                if isinstance(stream_, list):
+                    if not stream_:
+                        continue
+                    stream_ = stream_[0]
+
+                if ".m3u8" in stream_:
+                    for s in HLSStream.parse_variant_playlist(self.session, stream_).items():
+                        yield s
+                else:
+                    raise PluginError("Unexpected stream type: '{0}'".format(stream_))
```

Comment:
you can't **raise** something in a loop which might have valid streams

https://www.daserste.de/unterhaltung/film/filme-im-ersten/videos/mankells-wallander-eifersucht-video-102.html

Reply:
No, that’s right, because I had assumed that the plugin is only for live streams and there are no **mp4** streams, for example.

Reply:
if you use **raise** here, just **one invalid stream** can fail the whole plugin,
if you use a log message the plugin would still work for the **2nd valid stream**

**as an example,** when mp4 was not supported the plugin was broken even if there was a valid m3u8 url

Reply:
> as an example, when mp4 was not supported the plugin was broken even if there was a valid m3u8 url

It is important to me to eliminate mistakes and not to ignore them.